addons/sale/models/sale_order.py

```diff
@@ -1898,3 +1898,11 @@ def _is_paid(self):
         """
         self.ensure_one()
         return self.currency_id.compare_amounts(self.amount_paid, self.amount_total) >= 0
+
+    def _message_get_suggested_recipients(self):
+        recipients = super(SaleOrder, self)._message_get_suggested_recipients()
+        for order in self:
+            if order.partner_id:
+                reason = _('Customer Email') if order.partner_id.email else _('Customer')
+                order._message_add_suggested_recipient(recipients, partner=order.partner_id, reason=reason)
```

Comment:
I don't think you need to handle the different case, as you only handle the "partner_id" case of crm.lead, not the "email_from". Also, please use double quotes for visible/translatable strings. Lastly, keep the line below 100 characters.
```suggestion
                order._message_add_suggested_recipient(
                    recipients, partner=order.partner_id, reason=_("Customer")
                )
```

Reply:
I distinguished between customers with and without email addresses as that's what other models seem to do (e.g. [`project.task`](https://github.com/odoo/odoo/blob/d8a715b136b0bea4e68b00be6946873e06b2ee01/addons/project/models/project_task.py#L1494)), and `crm.lead`'s `email_from` in initialized with `partner_id.email`. I'm fine with either option since `reason` is probably not that important (it shows in a tooltip when you hover over the suggested customer, but I'm not aware of other uses). Do you have a preference?

Reply:
I agree that, with the example you provide in mind, I'd do the same. But in most other cases, the condition first checks whether the object has a partner, which is already te condition to arrive here, so I'd go for my suggestion (which is also shorter/simpler).

Reply:
Done :)